scripts/dts/extract/heuristics.py

```diff
@@ -39,10 +50,21 @@ def extract(self, node_address, yaml):
         # Process compatible related work
         try:
             compatible = reduced[node_address]['props']['compatible']
+            if isinstance(compatible, list):
+                compat = compatible[0]
+            else:
+                compat = compatible
         except KeyError:
             # No compat skip next part
             return
 
+        # Generat unique_id
+        unique_id = self._generate_string(node_address)
+        unique_id_str = unique_id.replace("-", "_").replace(",", "_"). \
+                                          replace("/","").upper()
+        edts_insert_device_property(node_address, 'unique_id', \
```

Comment:
Remove whole commit and use device-id where a unique ID for a device is needed.

Reply:
@b0661 , unique_id was a request from @galak , I let him answer on the need.

Why I implemented it that way:
- a)Not sure exactly on how `unique_id` will be used, so I didn't want to replace `device_id` with this `unique_id`. This is why it is implemented for now besides device_id and can be seen as redundant. I agree we can replace device_id with unique_id if we deem one of both is too many
- b)I thought about implementing a function that generate unique_id based on device_id as input. Though, still be cause I'm not sure about the future use, I chose to inject it directly in edts so i can be stored and always available afterwards, it was also easier to validate generated string was correct. I agree this could be done this way afterwards if having unique_id always available is not required
- c)I though about computing and injecting it within EDTSProviderMixin Class. Though, to generate unique_id string, it requires having access to parent information. EDTSProviderMixin class can only access information which is available in self._edts object, but at a time a node is computed, there is no guarantee its parent node is already populated. So I ended up implementing it in heuristics..
--> Btw, I think we should sort `reduced` in `generate_node_definitions`. Something like:
```
    oreduced  = collections.OrderedDict(sorted(reduced.items()))
    for k, v in oreduced.items():
```
 
Since we don't know about a) or b) yet, I think best option is to keep it that way for now.

Reply:
> Since we don't know about a) or b) yet, I think best option is to keep it that way for now.

Can´t you just leave it out as long as the usage is unclear and it is not used anyway?

It should be added when the usage is clear.

Reply:
To be checked with @galak

Reply:
I felt like we need this for codegen purposes.  We're going to want symbols generated that have some unique naming for a given device node.

Reply:
codegen accesses the device node info by "device-id" which is unique. Unique naming (e.g. for config_info, data, ...) is done using a combination of several device properties to generate a sanitized unique device name and append specifiers like  _data, _config, ... (see devicedeclare.py).

The goal was to help people to easily identify the device symbols in the generated code and the linker reports .

```
##
# @brief Get device name
#
# Device name is generated from
# - device compatible
# - bus master address if the device is connected to a bus master
# - device address
# - parent device address if the device does not have a device address
def _device_name(device_id):
    device_name = codegen.edts().get_device_property(device_id, 'compatible/0',
						     None)
    if device_name is None:
        codegen.error("No compatible property for device id '{}'."
                      .format(device_id))

    bus_master_device_id = codegen.edts().get_device_property(device_id,
							      'bus/master', None)
    if bus_master_device_id is not None:
        reg = codegen.edts().get_device_property(bus_master_device_id, 'reg')
        try:
            # use always the first key to get first address inserted into dict
            # because reg_index may be number or name
            # reg/<reg_index>/address/<address_index> : address
            for reg_index in reg:
                for address_index in reg[reg_index]['address']:
                    bus = reg[reg_index]['address'][address_index]
                    device_name += '_' + hex(bus)[2:].zfill(8)
                    break
                break
        except:
            # this device is missing the register directive
            codegen.error("No bus master register address property for device id '{}'."
                          .format(bus_master_device_id))

    reg = codegen.edts().get_device_property(device_id, 'reg', None)
    if reg is None:
        # no reg property - take the reg property of the parent device
        parent_device_id = codegen.edts().get_device_property(
            device_id, 'parent-device', None)
        if parent_device_id:
            reg = codegen.edts().get_device_property(parent_device_id, 'reg', None)
    device_address = None
    if reg is not None:
        try:
            # use always the first key to get first address inserted into dict
            # because reg_index may be number or name
            # reg/<reg_index>/address/<address_index> : address
            for reg_index in reg:
                for address_index in reg[reg_index]['address']:
                    address = reg[reg_index]['address'][address_index]
                    device_address = hex(address)[2:].zfill(8)
                    break
                break
        except:
            # this device is missing the register directive
            pass
    if device_address is None:
        # use label instead of address
        device_address = codegen.edts().get_device_property(device_id, 'label',
                                                        '<unknown>')
        # Warn about missing reg property
        codegen.log("No register address property for device id '{}'."
                    .format(device_id), "warning", "\n")
    device_name += '_' + device_address

    device_name = device_name.replace("-", "_"). \
                              replace(",", "_"). \
                              replace(";", "_"). \
                              replace("@", "_"). \
                              replace("#", "_"). \
                              replace("&", "_"). \
                              replace("/", "_"). \
                              lower()
    return device_name
```